File: Design_Pattern_and_Principles/ObserverPatternExample/ConcreteObservers.java
class MobileApp implements Observer {
    public void update(String stockSymbol, double price) {
        System.out.println("[Mobile App UI] Update received: " + stockSymbol + " is now $" + price);
    }
}

class WebApp implements Observer {
    public void update(String stockSymbol, double price) {
        System.out.println("[Web Dashboard] Synchronized: " + stockSymbol + " adjusted to $" + price);
    }
}

File: Design_Pattern_and_Principles/ObserverPatternExample/ConcreteObservers.cs
using System;

class MobileApp : IObserver {
    public void Update(string stockSymbol, double price) {
        Console.WriteLine("[Mobile App UI] Update received: " + stockSymbol + " is now $" + price);
    }
}

class WebApp : IObserver {
    public void Update(string stockSymbol, double price) {
        Console.WriteLine("[Web Dashboard] Synchronized: " + stockSymbol + " adjusted to $" + price);
    }
}